sound/soc/codecs/rt711-sdca.c

```diff
@@ -258,6 +258,8 @@ static int rt711_sdca_headset_detect(struct rt711_sdca_priv *rt711)
 	switch (det_mode) {
 	case 0x00:
 		rt711->jack_type = 0;
+		regmap_write(rt711->regmap,
+			SDW_SDCA_CTL(FUNC_NUM_JACK_CODEC, RT711_SDCA_ENT_GE49, RT711_SDCA_CTL_SELECTED_MODE, 0), 0);
```

Comment:
i am a bit confused by this patch @shumingfan 

Shouldn't the code override det_mode and change the value so that the updated selected_mode is written line 276?

Reply:
I am also confused by the rest of the switch case where the jack_type would be set based on the det_mode, not selected_mode.